thorlcr/activities/indexread/thindexreadslave.cpp

```diff
@@ -241,10 +232,27 @@ class CIndexReadSlaveBase : public CSlaveActivity
         keyIndexSet.setown(createKeyIndexSet());
         ForEachItemIn(p, partDescs)
         {
-            Owned<IKeyIndex> keyIndex = openKeyPart(partDescs.item(p));
-            Owned<IKeyManager> klManager = createKeyManager(keyIndex, fixedDiskRecordSize, NULL);
+            IPartDescriptor &part = partDescs.item(p);
+            RemoteFilename rfn;
+            part.getFilename(0, rfn);
+            StringBuffer filePath;
+            rfn.getPath(filePath);
+
+            Owned<IDelayedFile> lfile = queryThor().queryFileCache().lookup(*this, logicalFilename, part);
+            Owned<IKeyManager> klManager;
+
+            bool remoteKey = !localKey && !seekGEOffset && (!rfn.isLocal() || getOptBool("forceDafilesrv"));
```

Comment:
hm, because I've been testing locally and it is never enabled unless I force I had been thinking it was never enabled without forceDafilesrv... silly.

Yes, there should be an option to specifically enable. I'll add.